sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -190,3 +190,35 @@ def test_zero_division_hessians(data):
     X, y = data
     gb = HistGradientBoostingClassifier(learning_rate=100, max_iter=10)
     gb.fit(X, y)
+
+
+@pytest.mark.parametrize('GradientBoosting, data', [
+    (HistGradientBoostingClassifier,
+     make_classification(n_samples=10001, random_state=0, n_features=2,
+                         n_informative=2, n_redundant=0)),
+    (HistGradientBoostingRegressor,
+     make_regression(n_samples=10001, random_state=0, n_features=2,
+                     n_informative=2))]
+)
+def test_small_trainset(GradientBoosting, data):
+    # Make sure that a small trainset has the expected length (10k samples)
+    X, y = data
+    gb = GradientBoosting(random_state=42)
```

Comment:
I think you can merge this test into the other one, we don't really need to tests both classes. The classifier is enough (others might disagree?).

also nit: do you really need to set the random state of the estimator?

Reply:
Indeed, the random state of the estimator is not needed (I don't think that it is used, the dataset is small enough that the binning process will use the whole dataset).